docs/doxygen/man/man3/TransactionAccepted.3
.TH "TransactionAccepted" 3 "Thu Aug 30 2018" "esc" \" -*- nroff -*-
.ad l
.nh
.SH NAME
TransactionAccepted \- Transaction accepted TXSTYPE_UOK type\&.  

.SH SYNOPSIS
.br
.PP
.PP
\fC#include <transactionaccepted\&.h>\fP
.PP
Inherits \fBBlockCommand\fP\&.
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "\fBTransactionAccepted\fP (uint16_t abank, uint32_t auser, uint32_t amsid, uint32_t ttime, uint16_t bbank, uint32_t buser, uint8_t public_key[32])"
.br
.ti -1c
.RI "virtual int \fBgetType\fP () override"
.br
.RI "\fIGet type\&. \fP"
.ti -1c
.RI "virtual CommandType \fBgetCommandType\fP () override"
.br
.RI "\fIGet command type\&. \fP"
.ti -1c
.RI "virtual unsigned char * \fBgetData\fP () override"
.br
.RI "\fIGet pointer to command data structure\&. \fP"
.ti -1c
.RI "virtual unsigned char * \fBgetResponse\fP () override"
.br
.RI "\fIGet pointer to response data\&. \fP"
.ti -1c
.RI "virtual void \fBsetData\fP (char *data) override"
.br
.RI "\fIPut data as a char list and put convert it to data structure\&. \fP"
.ti -1c
.RI "virtual void \fBsetResponse\fP (char *response) override"
.br
.RI "\fIApply data as a response struct\&. \fP"
.ti -1c
.RI "virtual int \fBgetDataSize\fP () override"
.br
.RI "\fIGet data struct size\&. Without signature\&. \fP"
.ti -1c
.RI "virtual int \fBgetResponseSize\fP () override"
.br
.RI "\fIGet response data struct size\&. \fP"
.ti -1c
.RI "virtual unsigned char * \fBgetSignature\fP () override"
.br
.RI "\fIGet pointer to signature data\&. \fP"
.ti -1c
.RI "virtual int \fBgetSignatureSize\fP () override"
.br
.RI "\fIGet signature size\&. \fP"
.ti -1c
.RI "virtual void \fBsign\fP (const uint8_t *hash, const uint8_t *sk, const uint8_t *pk) override"
.br
.RI "\fISign actual data plus hash using user private and public keys\&. \fP"
.ti -1c
.RI "virtual bool \fBcheckSignature\fP (const uint8_t *hash, const uint8_t *pk) override"
.br
.RI "\fICheck actual signature\&. \fP"
.ti -1c
.RI "virtual uint32_t \fBgetTime\fP () override"
.br
.RI "\fIGet time of command\&. \fP"
.ti -1c
.RI "virtual uint32_t \fBgetUserId\fP () override"
.br
.RI "\fIGet User ID\&. \fP"
.ti -1c
.RI "virtual uint32_t \fBgetBankId\fP () override"
.br
.RI "\fIGet Node ID\&. \fP"
.ti -1c
.RI "virtual int64_t \fBgetFee\fP () override"
.br
.RI "\fIGet command fee\&. \fP"
.ti -1c
.RI "virtual int64_t \fBgetDeduct\fP () override"
.br
.RI "\fIGet change in cash balance after command\&. \fP"
.ti -1c
.RI "virtual uint32_t \fBgetUserMessageId\fP () override"
.br
.RI "\fIGet user message id\&. \fP"
.ti -1c
.RI "virtual bool \fBsend\fP (\fBINetworkClient\fP &netClient) override"
.br
.RI "\fISend data to the server\&. \fP"
.ti -1c
.RI "virtual void \fBsaveResponse\fP (\fBsettings\fP &sts) override"
.br
.RI "\fISave command response to settings object\&. \fP"
.ti -1c
.RI "virtual std::string \fBtoString\fP (bool pretty) override"
.br
.ti -1c
.RI "virtual void \fBtoJson\fP (boost::property_tree::ptree &ptree) override"
.br
.ti -1c
.RI "virtual void \fBtxnToJson\fP (boost::property_tree::ptree &ptree) override"
.br
.ti -1c
.RI "virtual std::string \fBusageHelperToString\fP () override"
.br
.in -1c
.SS "Additional Inherited Members"
.SH "Detailed Description"
.PP 
Transaction accepted TXSTYPE_UOK type\&. 
.PP
Definition at line 13 of file transactionaccepted\&.h\&.
.SH "Member Function Documentation"
.PP 
.SS "bool TransactionAccepted::checkSignature (const uint8_t * hash, const uint8_t * pk)\fC [override]\fP, \fC [virtual]\fP"

.PP
Check actual signature\&. 
.PP
\fBParameters:\fP
.RS 4
\fIhash\fP Previous hash operation\&. 
.br
\fIpk\fP Pointer to public key\&. 
.RE
.PP

.PP
Implements \fBICommand\fP\&.
.PP
Definition at line 60 of file transactionaccepted\&.cpp\&.
.SS "bool TransactionAccepted::send (\fBINetworkClient\fP & netClient)\fC [override]\fP, \fC [virtual]\fP"

.PP
Send data to the server\&. 
.PP
\fBParameters:\fP
.RS 4
\fInetClient\fP Netwrok client implementation of \fBINetworkClient\fP interface\&. 
.br
\fIpk\fP Pointer to public key\&. 
.RE
.PP

.PP
Implements \fBICommand\fP\&.
.PP
Definition at line 92 of file transactionaccepted\&.cpp\&.
.SS "void TransactionAccepted::sign (const uint8_t * hash, const uint8_t * sk, const uint8_t * pk)\fC [override]\fP, \fC [virtual]\fP"

.PP
Sign actual data plus hash using user private and public keys\&. 
.PP
\fBParameters:\fP
.RS 4
\fIhash\fP Previous hash operation\&. 
.br
\fIsk\fP Pointer to provate key\&. 
.br
\fIpk\fP Pointer to public key\&. 
.RE
.PP

.PP
Implements \fBICommand\fP\&.
.PP
Definition at line 56 of file transactionaccepted\&.cpp\&.

.SH "Author"
.PP 
Generated automatically by Doxygen for esc from the source code\&.


CHANGELOG.md
# Changelog
All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [Unreleased]

## [1.0.0] - 2018-08-24
### Changed
- Genesis file for the launch
- Update README

## [0.0.6] - 2018-08-10
### Added
- Profit sharing among TOP and VIP nodes

### Changed
- Genesis file for test launch

### Fixed
- Vote counting in get_block command
- Ability to send externally signed transactions

## [0.0.5] - 2018-08-01
### Added
- Support passing version by cmake command.
- Contributing guide
- "Dry run" test case

### Fixed
- Node stops after incorrect `send_many` request  - #134
- Fee checking
- Internal transactions numbers

## [0.0.4] - 2018-07-27
### Changed
- Update default parameters
- Creating a dev version with reference to the last tag

[Unreleased]: https://github.com/adshares/ads/compare/v1.0.0...HEAD

[1.0.0]: https://github.com/adshares/ads/compare/v0.0.6...v1.0.0
[0.0.6]: https://github.com/adshares/ads/compare/v0.0.5...v0.0.6
[0.0.5]: https://github.com/adshares/ads/compare/v0.0.4...v0.0.5
[0.0.4]: https://github.com/adshares/ads/compare/v0.0.3...v0.0.4


src/common/command/transactionaccepted.cpp
#include "transactionaccepted.h"

#include "abstraction/interfaces.h"
#include "helper/json.h"

TransactionAccepted::TransactionAccepted()
    : m_data{} {
    m_responseError = ErrorCodes::Code::eNone;
}

TransactionAccepted::TransactionAccepted(uint16_t abank, uint32_t auser, uint32_t amsid, uint32_t ttime, uint16_t bbank, uint32_t buser, uint8_t public_key[32])
    : m_data(abank, auser, amsid, ttime, bbank, buser, public_key) {
    m_responseError = ErrorCodes::Code::eNone;
}

unsigned char* TransactionAccepted::getData() {
    return reinterpret_cast<unsigned char*>(&m_data);
}

unsigned char* TransactionAccepted::getResponse() {
    return reinterpret_cast<unsigned char*>(&m_response);
}

void TransactionAccepted::setData(char* data) {
    m_data = *reinterpret_cast<decltype(m_data)*>(data);
}

void TransactionAccepted::setResponse(char* response) {
    m_response = *reinterpret_cast<decltype(m_response)*>(response);
}

int TransactionAccepted::getDataSize() {
    return sizeof(m_data);
}

int TransactionAccepted::getResponseSize() {
    return 0;
}

unsigned char* TransactionAccepted::getSignature() {
    return (unsigned char*)"";
}

int TransactionAccepted::getSignatureSize() {
    return 0;
}

int TransactionAccepted::getType() {
    return TXSTYPE_UOK;
}

CommandType TransactionAccepted::getCommandType() {
    return CommandType::eReadingOnly;
}

void TransactionAccepted::sign(const uint8_t* /*hash*/, const uint8_t* /*sk*/, const uint8_t* /*pk*/) {
    // no signature
}

bool TransactionAccepted::checkSignature(const uint8_t* /*hash*/, const uint8_t* /*pk*/) {
    // no signature
    return true;
}

void TransactionAccepted::saveResponse(settings& /*sts*/) {
}

uint32_t TransactionAccepted::getUserId() {
    return 0;
}

uint32_t TransactionAccepted::getBankId() {
    return 0;
}

uint32_t TransactionAccepted::getTime() {
    return 0;
}

int64_t TransactionAccepted::getFee() {
    return 0;
}

int64_t TransactionAccepted::getDeduct() {
    return 0;
}

uint32_t TransactionAccepted::getUserMessageId() {
    return 0;
}

bool TransactionAccepted::send(INetworkClient& /*netClient*/) {
    return true;
}

std::string TransactionAccepted::toString(bool /*pretty*/) {
    return "";
}

void TransactionAccepted::toJson(boost::property_tree::ptree& /*ptree*/) {
}

void TransactionAccepted::txnToJson(boost::property_tree::ptree& ptree) {
    using namespace Helper;
    ptree.put(TAG::TYPE, getTxnName(m_data.ttype));
    ptree.put(TAG::SRC_NODE, m_data.abank);
    ptree.put(TAG::SRC_USER, m_data.auser);
    ptree.put(TAG::MSGID, m_data.amsid);
    ptree.put(TAG::TIME, m_data.ttime);
    ptree.put(TAG::DST_NODE, m_data.bbank);
    ptree.put(TAG::DST_USER, m_data.buser);
    ptree.put(TAG::PKEY, ed25519_key2text(m_data.publicKey, sizeof(m_data.publicKey)));
}

std::string  TransactionAccepted::usageHelperToString() {
    return std::string("");
}
